crates/services/txpool_v2/src/pool.rs

```diff
@@ -236,9 +255,23 @@ where
         &mut self,
         constraints: Constraints,
     ) -> Vec<ArcPoolTx> {
-        self.selection_algorithm
-            .gather_best_txs(constraints, &mut self.storage)
+        let start = std::time::Instant::now();
+        let metrics = self.config.metrics;
```

Comment:
```rust
let start = if self.config.metrics {
    Some(std::time::Instant::now())
} else {
    None
};

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2385/commits/da77cf99329aec61abdbefb713d263248cd4f78c